lib/ansible/modules/monitoring/sensu_silence.py

```diff
@@ -0,0 +1,387 @@
+            module, url, method='POST',
+            headers=headers, data=json.dumps(request_data)
+        )
+
+        if info['status'] != 204:
+            module.fail_json(
+                msg="Failed to silence %s. Reason: %s" % (subscription, info)
+            )
+
+        try:
+            json_out = json.loads(response.read())
+        except:
+            json_out = ""
+
+        return False, json_out, True
+    return False, out, True
+
+
+def create(
+        module, url, check, creator, expire,
+        expire_on_resolve, reason, subscription):
+    (rc, out, changed) = query(module, url, check, subscription)
+    for i in out:
+        if (i['subscription'] == subscription):
+            if (
+                    (check is None or check == i['check']) and
+                    (
+                        creator == 'Created via Ansible' or
+                        creator == i['creator'])and
+                    (
+                        reason == 'No reason given, tisk tisk' or
+                        reason == i['reason']) and
+                    (
+                        expire is None or expire == i['expire']) and
+                    (
+                        expire_on_resolve is None or
+                        expire_on_resolve == i['expire_on_resolve']
+                    )
+               ):
+                return False, out, False
+
+    # module.check_mode is inherited from the AnsibleMOdule class
+    if not module.check_mode:
+        headers = {
+            'Content-Type': 'application/json',
+        }
+
+        url = url + '/silenced'
+
+        request_data = {
+            'check': check,
+            'creator': creator,
+            'expire': expire,
+            'expire_on_resolve': expire_on_resolve,
+            'reason': reason,
+            'subscription': subscription,
+        }
+
+        # Remove keys with None value
+        for k, v in dict(request_data).items():
+            if v is None:
+                del request_data[k]
+
+        response, info = fetch_url(
+            module, url, method='POST',
+            headers=headers, data=json.dumps(request_data)
+        )
+
+        if info['status'] != 201:
+            module.fail_json(
+                msg="Failed to silence %s. Reason: %s" %
+                (subscription, info['msg'])
+            )
+
+        try:
+            json_out = json.loads(response.read())
+        except:
+            json_out = ""
+
+        return False, json_out, True
+    return False, out, True
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            check=dict(required=False),
+            creator=dict(required=False, default='Created via Ansible'),
+            expire=dict(required=False),
+            expire_on_resolve=dict(required=False),
+            reason=dict(required=False, default='No reason given, tisk tisk'),
+            state=dict(default='create', choices=['create', 'clear']),
```

Comment:
Thanks for all these comments, I'll look into these hopefully tomorrow or Friday